### PR TITLE
Add observable pins per cluster

### PR DESCRIPTION
**Pull request by rgantonio**
This PR adds observable pins to each SNAX cluster. (This has been adapted according to @IveanEx's suggestion).

We have observable pins only from the first snitch core as shown below:

image

Details:
* It is an RW register inside the Snitch core and part of the Snitch CSR registers.
* The register address is `0x7c5` which was arbitrarily chosen. It can change later. It occupies a free CSR space.
* How do we use it?
  * It's a simple RW register, it functions like a SW debug for observability.
  * In the C-code (or any code) you can insert CSR write operations to write unto this register.
  * Suppose your accelerator has states: idle > run > busy. Then you put in the C-code, some arbitrary value (say you want `0xffa` to indicate the idle mode. Then when you activate your accelerator, write `0xffb` to the register. 
  * Note that it is only connected to the 1st core. All other cores do have the register but they are not visible to the top. Just the first core.

Major TODO:
- [x] Modify Snitch
- [x] Modify Snitch core cluster
- [x] Modify Snitch cluster
- [x] Modify Snitch cluster template
- [x] Add switch in config signals
- [x] Add csr run time to read or write
- [x] Test if it works by checking if the signals are visible from the top cluster

### CODE REVIEW

**Comment by IveanEx** [comment]
May I ask why not connect the observation pin to one of the csr in snitch core but connect it to the accelerator 🫣🫣

**Comment by rgantonio** [comment]
> May I ask why not connect the observation pin to one of the csr in snitch core but connect it to the accelerator 🫣🫣To avoid the case that UART for some reasons does not work

I'm not sure I understood, can you please draw if possible? What do you mean by being one of the CSRs in snitch core and then connecting to the accelerator? And how does it help in case UART does not work? 

I originally thought that it would be nice if we just provided pins but allowed people to use the pins any way they want to.

**Comment by IveanEx** [comment]
> > May I ask why not connect the observation pin to one of the csr in snitch core but connect it to the accelerator 🫣🫣To avoid the case that UART for some reasons does not work
> 
> @IveanEx
> 
> I'm not sure I understood, can you please draw if possible? What do you mean by being one of the CSRs in snitch core and then connecting to the accelerator? And how does it help in case UART does not work?
> 
> I originally thought that it would be nice if we just provided pins but allowed people to use the pins any way they want to.

For instance, just create one CSR at the first core of each cluster and put 4 LSBs to the outside for instance. (Irrelevant to accelerator, just the core)

**Comment by rgantonio** [comment]
> For instance, just create one CSR at the first core of each cluster and put 4 LSBs to the outside for instance. (Irrelevant to accelerator, just the core)

Is that something you want? We can add that too.

I was originally thinking that there may be bits that are not just simple RW registers like a simple RW CSR in Snitch. We can do that in Snitch sure, but it would also be nice to have a mechanism, where from the accelerators themselves, they give out status signals that are sticky.

For example, the (1) part I had was that my upper 2 MSB are state-dependent inside my accelerator CSRs. When I kick start for my accelerator, the 2 MSB becomes 2'b01. The state changes depending on what my accelerator is doing.

BUT! That is just because I want to debug my accelerator that way if that makes sense?

If I just do a simple CSR from the Snitch core (which could also be possible btw) but it's just as if I'm looking at the Snitch core and not really inside the accelerator.

Note that anyone can do what they want though. We're just giving options of debug ports to what each cluster person can use! So it's just like "Hey, here are some ports you can connect stuff for observability. We'll leave it up to you on what to check".

**Comment by rgantonio** [comment]
@IveanEx! I have reflected upon this and what your suggesting might be "easier" for everyone. But we discuss first when you're in the office 😄

**Comment by rgantonio** [comment]
Bump @IveanEx

I made it as how you described it yesterday. I limited it to 8 bits for now but can be expanded later. We need to limit all pins to 132 pins for the current tapeout.

**Comment by rgantonio** [comment]
Anyone care to approve if this is okay? haha

**Review by xiaoling-yi — APPROVED** [review]
Ok, thank you @rgantonio !

**Pull request merged**
